src/components/Select/index.js

```diff
@@ -1,36 +1,172 @@
+                    type="text"
+                    key={`input${innerValue}`}
+                    value={versionName}
+                    onChange={handleChangeVersionName}
+                    placeholder={innerValue}
+                    onKeyDown={(e) => {
+                      handleKeyDown(innerValue, iconsAndFunctions, index, e);
+                    }}
+                  />
+                ) : (
+                  <Button
+                    className="option-text"
+                    key={innerValue}
+                    onClick={() => onClick(innerValue)}
+                  >
+                    {iconsAndFunctions.icons.includes('add') ? (
+                      <FontAwesomeIcon fixedWidth icon={faPlus} />
+                    ) : null}
+                    {innerLabel}
+                  </Button>
+                )}
+                {iconsAndFunctions.icons.includes('edit') ? (
+                  <Button
+                    key={`${innerValue}edit`}
+                    className="option-button"
+                    onClick={(e) => {
+                      if (innerValue === inputIndex) {
+                        if (versionName === '') {
+                          setVersionName('Blank');
+                        }
+                        if (iconsAndFunctions.functions.edit(versionName)) {
+                          setCurrentValue(
+                            innerValue === currentValue
+                              ? versionName
+                              : currentValue
+                          );
+                          optionsObtained[index].innerValue = versionName;
+                          optionsObtained[index].innerLabel = versionName;
+                        }
+                        setInputting(false);
+                        setInputIndex('');
+                      } else {
+                        setVersionName(innerValue);
+                        setInputting(true);
+                        setInputIndex(innerValue);
+                        e.stopPropagation();
+                      }
+                    }}
+                  >
+                    <FontAwesomeIcon fixedWidth icon={faEdit} />
+                  </Button>
+                ) : null}
+                {iconsAndFunctions.icons.includes('delete') ? (
+                  <Button
+                    key={`${innerValue}delete`}
+                    className="option-button"
+                    onClick={() => {
+                      swal({
+                        buttons: ['Cancel', 'Delete'],
+                        className: `${theme}`,
+                        content: (
+                          <p>
+                            {/* eslint-disable-next-line max-len */}
+                            {/* eslint-disable-next-line react/no-unescaped-entities */}
+                            Are you sure you want to delete "{innerLabel}
+                            {'" '}
+                            schedule?
```

Comment:
Again, this is a very specific reference to the schedule version delete behavior in a general-purpose component

Reply:
Extracted this swal alert out to the delete function passed into the general <Select /> component. Everything in <Select /> should be general and decoupled from version select. Each option in <Select /> takes in its identifier(optionId), its text(optionLabel), its onClick, its related icons and their corresponding functions. However, each icon's display (before or after the text, when to display), picture have to be specifed within the <Select />.